### PR TITLE
ValuesSource refactoring: Wire up SigTerms aggregation

### PR DESCRIPTION
**Pull request by polyfractal**
Pretty similar to Terms agg, albeit simpler because SigTerms has fewer options.  

I opted to _not_ try and refactor the "singleton factory" functionality, it proved more complicated than I wanted to deal with in this PR.  Just left a TODO so we can work on it unrelated to VS stuff in the future.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregatorFactory.java [hunk 1]
```diff
@@ -75,17 +78,116 @@
...
+                }
+                if (valuesSource instanceof ValuesSource.Bytes.WithOrdinals == false) {
+                    execution = ExecutionMode.MAP;
+                }
+                if (execution == null) {
+                    execution = ExecutionMode.GLOBAL_ORDINALS;
+                }
+
+                if ((includeExclude != null) && (includeExclude.isRegexBased()) && format != DocValueFormat.RAW) {
+                    throw new IllegalArgumentException("Aggregation [" + name + "] cannot support regular expression style "
```
Comment (polyfractal) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregatorFactory.java]: Just noticed I forgot to change this like we've been doing on others, will fix in the morning

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregatorFactory.java [hunk 2]
```diff
@@ -75,17 +78,116 @@
...
+                                    String executionHint,
+                                    SearchContext context,
+                                    Aggregator parent,
+                                    SignificanceHeuristic significanceHeuristic,
+                                    SignificantTermsAggregatorFactory sigTermsFactory,
+                                    List<PipelineAggregator> pipelineAggregators,
+                                    Map<String, Object> metaData) throws IOException {
+
+                if ((includeExclude != null) && (includeExclude.isRegexBased())) {
+                    throw new AggregationExecutionException("Aggregation [" + name + "] cannot support regular expression style include/exclude "
```
Comment (polyfractal) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregatorFactory.java]: Just noticed I forgot to change this, should be IllegalArgException, will fix in the morning :)

**Review by not-napoleon — APPROVED** [review]
LGTM

**Review by csoulios — COMMENTED** [review]
LGTM

**Pull request merged**